### PR TITLE
fix(ci): Create separate testing Dockerfile

### PR DESCRIPTION
**Pull request by evanh**
Create a separate testing Dockerfile so we don't ship images with the test
requirements to prod/self hosted.

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -9,14 +9,25 @@ steps:
   args: [
             'build',
             '-t', 'us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA',
+            '-t', 'getsentry/snuba:ci',
```
Comment (BYK) [inline cloudbuild.yaml:12]: Any reason why we are tagging this as `ci`? We should be able to pass `COMMIT_SHA` as a build arg and parametrize the `Dockerfile.testing` file as follows:

```dockerfile
ARG COMMIT_SHA
FROM getsentry/snuba:${COMMIT_SHA}

RUN pip install -r requirements-test.txt
```

**Comment by evanh** [comment]
@BYK Is there a way to test to make sure this is now creating the correct production image?

File: Dockerfile [hunk 2]
```diff
@@ -95,7 +95,3 @@ ENV SNUBA_RELEASE=$SNUBA_VERSION_SHA \
 EXPOSE 1218
 ENTRYPOINT [ "./docker_entrypoint.sh" ]
 CMD [ "api" ]
-
-FROM application AS testing
```
Comment (BYK) [inline Dockerfile:99]: I think we can also remove the `AS application` from the top

**Review by BYK — APPROVED** [review]
LGTM

**Comment by BYK** [comment]
>@BYK Is there a way to test to make sure this is now creating the correct production image?

You should be able to pull `us.gcr.io/sentryio/snuba:a44df9dd7c5882ff56a3784c48394a77b1a4a7cc` and check that image which should be built by GCB now.

**Comment by evanh** [comment]
I pulled that image and confirmed that the test requirements are not installed.

**Review by fpacifici — APPROVED** [review]

**Comment by evanh** [comment]
@BYK I have no idea how to make this work with the cloudbuild changes you just added. Any suggestions on how to split the main and testing images now?

**Comment by BYK** [comment]
@evanh ehh, sorry. I'll take this over and revise.

**Comment by evanh** [comment]
@BYK is going to take a crack at this.

**Pull request closed without merging**